docs/concepts/tools/kubectl/object-management-using-declarative-config.md

```diff
@@ -0,0 +1,517 @@
+{% endcomment %}
+
+
+### Merging changes to primitive fields
+
+Changes to a primitive field of an object use the following strategy:
+
+| Field in local object configuration | Field in live object configuration | Field in last-applied-config | Action                                    |
+|-------------------------------------|------------------------------------|------------------------------|-------------------------------------------|
+| Yes                                 | Yes                                | -                            | Conflict! Set live to local configuration |
+| Yes                                 | No                                 | -                            | Set live to local configuration           |
+| No                                  | -                                  | Yes                          | Delete from live configuration            |
+| No                                  | -                                  | No                           | Do nothing / Keep live value              |
+
+### Merging changes to map (object) fields
+
+Changes to map fields of an object use the following strategy for each key in the map:
+
+| Key in local object configuration   | Key in live object configuration   | Field in last-applied-config | Action                           |
+|-------------------------------------|------------------------------------|------------------------------|----------------------------------|
+| Yes                                 | Yes                                | -                            | Recurse on map value             |
+| Yes                                 | No                                 | -                            | Set live to local configuration  |
+| No                                  | -                                  | Yes                          | Delete from live configuration   |
+| No                                  | -                                  | No                           | Do nothing / Keep live value     |
+
+### Merging changes for list fields
+
+Merging changes to lists uses one of 2 strategies.  The strategy is controlled
+per-field on the Kubernetes object definition.
+
+- Replace: Treat the list the same as a primitive field.  Replace or delete the entire list.  Preserves ordering.
+- Merge: Treat the list like a map.  Add, delete, or update individual entries.  Does not preserve ordering.
+
+The merge strategy requires that a `mergeKey` be defined on the list field.
+The `mergeKey` is a field on the list elements to be used as the map key when
+merging the list as though it were a map.
+
+**Note:** As of Kubernetes 1.5, lists of primitive types do not support the merge strategy.
+
+{% comment %}
+TODO(1.6): Update this with the following for 1.6
+
+Lists of primitive types with the merge strategy use the value as the `mergeKey`.
```

Comment:
It is lost. Lists of primitive types with the merge strategy behave as a set as stated in [api-sonvention](https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#unordered-set)